contracts/TradetrustERC721.sol

```diff
@@ -74,28 +86,67 @@ contract TradeTrustERC721 is TitleEscrowCloner, ERC721Mintable, IERC721Receiver
     return this.onERC721Received.selector;
   }
 
-  function destroyToken(uint256 _tokenId) public onlyMinter {
-    require(ownerOf(_tokenId) == address(this), "Cannot destroy token: Token not owned by token registry");
+  function _burnToken(uint256 _tokenId) internal onlyMinter {
     // Burning token to 0xdead instead to show a differentiate state as address(0) is used for unminted tokens
     emit TokenBurnt(_tokenId);
     this.safeTransferFrom(ownerOf(_tokenId), 0x000000000000000000000000000000000000dEaD, _tokenId, "");
   }
 
+  function acceptSurrender(uint256 tokenId) external onlyMinter {
+    require(ownerOf(tokenId) == address(this), "TokenRegistry: Token has not been surrendered");
+    _burnToken(tokenId);
+  }
+
   // TODO: modify this to mint a new token if it doesn't exist, and send a token if it is owned by address(this)
   // rationale for this is that we currently also have a need for a method that performs the action for
   // minting to a new title escrow directly, which has a large overlap with this function
   // make sure to write tests for this and check for access controls
+  // Question: Should we combine the 2 actions into a single function or split them into their own functions instead?
```

Comment:
i'm tending towards single function but its not a strong preference

Reply:
If split `sendToNewTitleEscrow` into 2 separate functions, ie, `restoreTitle` and `mintTitle`, we can actually don't need the function `sendToNewTitleEscrow` anymore right? 
It seems to be originally used for minting a new title escrow and we can just call `mintTitle()`.

https://github.com/Open-Attestation/token-registry/blob/7a6deb6df09fa37560a0114e387313e85618673a/contracts/TradetrustERC721.sol#L135-L146

Reply:
indeed, i thought you just both options in for illustration purposes